### DISCUSSION TITLE
Install Onedrive 2.5.2 and OneDrive GUI 1.1.1

### DISCUSSION THREAD

**Discussion by moijeux1** [abraunegg/onedrive#2960] [Q&A] [answered]
I had have a lot of difficult to instal the new version of OneDrive 2.5.2-1. 

Thank you first for the work you have done because even if I have a lot of problems with apps, I recognize that there are actions behind it, time that has been taken for the good of everyone.
I try to read the documentation but since it is in English and I have trouble with this language (despite the fact that I am learning it at the moment ;-)), it is quite difficult. Even if I translate it, you probably know that a translation is not worth the original especially when there are technical terms.
I finally managed (apparently) to make it work. I went to get the OneDrive 2.5.2-1 application here:
https://packages.debian.org/sid/onedrive.
I got the OneDriveGUI 1.1.1 app:
https://github.com/bpozdena/OneDriveGUI ... ag/v1.1.1a
I obviously deleted the old OneDrive apps:
```
sudo apt remove onedrive
sudo apt update
sudo apt upgrade
sudo apt autoremove
sudo apt autoremove --purge
```
I guess what I did is not good but in itself, I think it is not serious, I probably did useless things.

I went to search in the "file system" file everything related to Onedrive to delete what was not deleted elsewhere. I found configuration files, backup files and.
Then I installed the OneDrive 2.5.2-1 deb file
I configured the onedrive1.1.1 AppImage (permission)
And Oh miracle !  I managed to configure the client normally to get the synchronization.

New problem : The client application (onedrive 1.1.1) gets stuck at : OneDrive  is processing Items... Nothing move, nothing work.

**Comment by moijeux1** [comment]
Remake operation and finally... that's working...for the instant

**Comment by abraunegg** [comment, marked as answer]
@moijeux1 
All of the steps you followed are *not* correct and will lead to future trouble.

For *all* Debian and Ubuntu platforms you must install the application following these instructions: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

For example:

#### Step 1: Remove old version
```
sudo apt remove onedrive
```

#### Step 2: Determine your distribution
```
lsb_release -a
```

#### Step 3: Based on output of 'Step 2' follow the instructions for the distribution

Read: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#step-4-pick-the-correct-instructions-to-use

#### Step 4: Follow the distribution details
You are using Linux Mint:

![image](https://github.com/user-attachments/assets/eb12e12d-cc73-4d57-91f3-a2a2ea42b842)

#### Step 5: Example using Linux Mint 22.x following Ubuntu 24.04 instructions:
```
wget -qO - https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_24.04/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/obs-onedrive.gpg > /dev/null
echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/obs-onedrive.gpg] https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_24.04/ ./" | sudo tee /etc/apt/sources.list.d/onedrive.list
sudo apt-get update
sudo apt install --no-install-recommends --no-install-suggests onedrive
```


And that is all there is to it ... 4 commands to install the client correctly on your platform

> New problem : The client application (onedrive 1.1.1) gets stuck at : OneDrive is processing Items... Nothing move, nothing work.

This is *not* a problem ... it is processing your data ...... please *read* the documentation

**Reply by moijeux1** [reply]
The problem is that I tried to do as you describe in the documentation and in two months (and even more) I never managed to make the application work. So I had to try by myself and based on what I read elsewhere.
I reinstalled Linux Mint 2 or 3 times because I thought it was the installation that was causing the problem. When it seemed to work, I realized that the synchronization was done in two different files. And when I tried to fix it, 99% of my files had disappeared from OneDrive (the server). So I had to start all over again, discovered this solution that worked.
If I still have a problem, I will start again by going over the steps and taking screenshots of each step and especially the error steps and I will post it here so that you can tell me where I made a mistake!

**Comment by moijeux1** [comment]
Hello, I don't know about others but no matter how I do it, it doesn't work in the long term.
At one point, some files downloaded from OneDrive (the server) are no longer readable (there is the image of the file but it is impossible to access it). I reinstalled OneDrive 2.5-2 in the way you give: The first time it worked, except that once my computer rebooted, no more synchronization, the 1.1.1 client starts up then stops immediately. Last reinstallation, this time the client does not even see OneDrive installed (while obviously it is the case).

**Comment by abraunegg** [comment]
> Hello, I don't know about others but no matter how I do it, it doesn't work in the long term.
At one point, some files downloaded from OneDrive (the server) are no longer readable (there is the image of the file but it is impossible to access it). I reinstalled OneDrive 2.5-2 in the way you give: The first time it worked, except that once my computer rebooted, no more synchronization, the 1.1.1 client starts up then stops immediately. Last reinstallation, this time the client does not even see OneDrive installed (while obviously it is the case).


You are most likely doing something *wrong* - there is absolutely zero reason to keep re-installing the application.

Please re-read all the documentation very carefully.

Once the client is installed and authenticated, the client *should* just work.

Ignoring the OneDrive GUI 1.1.1 here for a moment (and for issues with that you need to raise your issue or question here: https://github.com/bpozdena/OneDriveGUI) - you need to ensure the client is operating correctly from a terminal session before doing anything else.

#### Tips
1. Read the documentation - and potentially use Google Translate to translate the documents to French to assist you
2. Get the client to work / operate as per how you are wanting it to work via the CLI. The GUI is nothing more than a wrapper for the CLI operation
3. Add `--dry-run` to test your configuration. Add `--verbose` to understand what the client is doing at a more detailed level.

#### Example usage:

What does your client do, when you drop to a command prompt / terminal session for each of these commands:
```
onedrive --display-config
```
```
onedrive --sync --verbose
```

> The first time it worked, except that once my computer rebooted, no more synchronization

This is because you need to configure a systemd service.

This is where the OneDrive GUI assists, or you can manually set this up by reading https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#how-to-initiate-the-client-as-a-background-service
